sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/WritableColumnVector.java

```diff
@@ -350,6 +355,14 @@ public WritableColumnVector reserveDictionaryIds(int capacity) {
    */
   public abstract void putArray(int rowId, int offset, int length);
 
+  /**
+   * Puts a new non-null struct at 'rowId' of this vector, which is backed by elements at
+   * 'offset' of child vectors.
+   *
+   * NOTE: this MUST be called after new elements are appended to child vectors of a struct vector.
+   */
+  public abstract void putStruct(int rowId, int offset);
```

Comment:
I want to point out that this is a behavior change in `WritableColumnVector` so anyone who's using this to write struct will need to adopt to the new API. 

Wanna see if you have any concern on this @cloud-fan @sadikovi .

Reply:
IMHO, it is fine as the PR goes to master, this is supposed to be internal API anyway unless I am mistaken 🙂.

Reply:
yea `WritableColumnVector` is an internal API

Reply:
cool, thanks for confirming!